### PR TITLE
rest of maps homework

### PR DESCRIPTION
**Pull request by Evgenygroup**
#38

### CODE REVIEW

File: src/homework_collectionsassignments_maps/HomeworkMaps.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    Примеры:
+    mapAB({"a": "Hi", "b": "There"})  → {"a": "Hi", "ab": "HiThere", "b": "There"}
+    mapAB({"a": "Hi"}) → {"a": "Hi"}mapAB({"b": "There"}) → {"b": "There"}
+
+    */
+
+    public Map<String, String> ab(Map<String, String> input) {
+        Map<String, String> mapAB = new HashMap();
+        if (input.containsKey("a")) mapAB.put("a", input.get("a"));
+        if (input.containsKey("b")) mapAB.put("b", input.get("b"));
```
Comment (sergeylukichev) [inline src/homework_collectionsassignments_maps/HomeworkMaps.java:20]: I am not sure this is correct. If a key with a value is already there, makes no sense to put it again with the same value.

File: src/homework_collectionsassignments_maps/HomeworkMaps.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        if (input.containsKey("a") && input.containsKey("b")) mapAB.put("ab", input.get("a").concat(input.get("b")));
```
Comment (sergeylukichev) [inline src/homework_collectionsassignments_maps/HomeworkMaps.java:21]: This like seems to be enough for the solution.

**Review by sergeylukichev — APPROVED** [review]
👍

File: src/homework_collectionsassignments_maps/HomeworkMaps.java (hunk 1 above)
Comment (Evgenygroup) [reply src/homework_collectionsassignments_maps/HomeworkMaps.java:20]: But they are for different maps -the idea is if there is only key a or only key b in initial map, then key a or key b are nonetheless created in mapAB. Judging by the task examples it is so:
mapAB({"a": "Hi"}) → {"a": "Hi"};  mapAB({"b": "There"}) → {"b": "There"} If the new value is only created when a and b are present in initial map, then of course  these operations are redundant.

**Pull request merged**